be/src/exec/hdfs_scanner.cpp

```diff
@@ -433,33 +425,21 @@ StatusOr<bool> HdfsScannerContext::should_skip_by_evaluating_not_existed_slots()
 }
 
 void HdfsScannerContext::append_or_update_partition_column_to_chunk(ChunkPtr* chunk, size_t row_count) {
-    if (partition_columns.size() == 0 || row_count <= 0) return;
-
+    if (partition_columns.size() == 0 || row_count < 0) return;
     ChunkPtr& ck = (*chunk);
-    ck->set_num_rows(row_count);
-
     for (size_t i = 0; i < partition_columns.size(); i++) {
         SlotDescriptor* slot_desc = partition_columns[i].slot_desc;
         DCHECK(partition_values[i]->is_constant());
-        auto* const_column = ColumnHelper::as_raw_column<ConstColumn>(partition_values[i]);
-        ColumnPtr data_column = const_column->data_column();
-        auto chunk_part_column = ColumnHelper::create_column(slot_desc->type(), slot_desc->is_nullable());
-
-        if (row_count > 0) {
-            if (data_column->is_nullable()) {
-                chunk_part_column->append_nulls(1);
-            } else {
-                chunk_part_column->append(*data_column, 0, 1);
-            }
-            chunk_part_column->assign(row_count, 0);
-        }
-
+        // create constant column here.
+        ColumnPtr chunk_part_column = partition_values[i]->clone_shared();
```

Comment:
in this way, chunk part column is const column but with size as 1, so we have to resize it next.